src/Stratis.Bitcoin.Features.Consensus/PosConsensusValidator.cs

```diff
@@ -135,6 +135,8 @@ public override void CheckBlock(ContextInformation context)
             // Check timestamp.
             if (block.Header.Time > this.FutureDrift(this.dateTimeProvider.GetAdjustedTimeAsUnixTimestamp()))
             {
+                // The block can be valid only after its time minus the future drift.
+                context.BlockValidationContext.RejectUntil = Utils.UnixTimeToDateTime(block.Header.Time - this.FutureDrift(0)).UtcDateTime;
```

Comment:
`block.Header.Time` is already in UTC no? do we need `.UtcDateTime`

Reply:
Ah `Utils.UnixTimeToDateTime` is DateTimeOffset ignore my comment.

Reply:
I am not sure if this code should be here or in the rejected if condition, putting it here requires to add `RejectUntil` to the context this can be avoided.

Reply:
elsewhere we don't have FutureDrift, that's why it is here

Reply:
Ah good point, very well lets push it.